### PR TITLE
[FIX] sale_project: prevent unacceptable project form loading time

### PR DESCRIPTION
**Pull request by IT-Ideas**
Previous implementation was generating an unoptimised query which
lead to unacceptable loading time on large databases.

task-2697814

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_project/models/project.py [hunk 1]
```diff
@@ -36,12 +36,26 @@ def _compute_sale_line_id(self):
     @api.depends('sale_order_id.invoice_status', 'tasks.sale_order_id.invoice_status')
     def _compute_has_any_so_to_invoice(self):
         """Has any Sale Order whose invoice_status is set as To Invoice"""
-        project_to_invoice = self.env['project.project'].search([
-            '|',
-                ('sale_order_id.invoice_status', '=', 'to invoice'),
-                ('tasks.sale_order_id.invoice_status', '=', 'to invoice'),
-            ('id', 'in', self.ids)
-        ])
+        if not len(self.ids):
```
Comment (madprog) [inline addons/sale_project/models/project.py:39]: What about this?
```suggestion
        if not self:
```

Comment (IT-Ideas) [reply addons/sale_project/models/project.py:39]: The problem I had came when the record is not created yet and that ids is empty as it only contains a new_id.
This is a test that crashed.
So I think self still exists, its only ids that is empty which leads to an incorrect query, isn't it?

Comment (IT-Ideas) [reply addons/sale_project/models/project.py:39]: I'll give it a try... Thanks 🙂

Comment (madprog) [reply addons/sale_project/models/project.py:39]: Oh, if it is about new_ids, then you should probably call `self.flush()`, but I'm not sure about the implications to call it in a compute method :fearful:

Comment (IT-Ideas) [reply addons/sale_project/models/project.py]: Never work too late... I did forget the return 🙈

**Comment by IT-Ideas** [comment]
@robodoo r+

**Pull request closed without merging**